common/scala/src/main/scala/whisk/core/database/DocumentFactory.scala

```diff
@@ -245,26 +261,35 @@ trait DocumentFactory[W] extends MultipleReadersSingleWriterCache[W, DocInfo] {
     }
   }
 
-  def getAttachment[Wsuper >: W](db: ArtifactStore[Wsuper],
-                                 doc: DocInfo,
-                                 attachmentName: String,
-                                 outputStream: OutputStream)(implicit transid: TransactionId): Future[Unit] = {
+  def getAttachment[Wsuper >: W](
+    db: ArtifactStore[Wsuper],
+    entity: W,
+    doc: DocInfo,
+    attachmentName: String,
+    outputStream: OutputStream,
+    postProcess: Option[PostProcessEntity[W]] = None)(implicit transid: TransactionId, mw: Manifest[W]): Future[W] = {
 
     implicit val ec = db.executionContext
+    implicit val notifier: Option[CacheChangeNotification] = None
```

Comment:
Cache invalidation callback should probably be assigned to the actual callback here.